### ISSUE TITLE
Mining misssion cannot be accomplished

### ISSUE THREAD

**Issue by Gaulandm** [Regalis11/Barotrauma#6640]
The generation makes it impossible to reach the uranium and mine it.
https://prnt.sc/1rx663m

**Comment by Regalis11** [comment]
I can't see any unreachable areas in the pic you posted - this may be a dumb question, but are you aware that you can break those translucent ice chunks with plasma cutters and explosives?

Would you still happen to have the save where this happened? The issue might be impossible for us to reproduce without a save or a level seed.

Thanks and sorry for the inconvenience!

**Comment by Gaulandm** [comment]
> I can't see any unreachable areas in the pic you posted - this may be a dumb question, but are you aware that you can break those translucent ice chunks with plasma cutters and explosives?
> 
> Would you still happen to have the save where this happened? The issue might be impossible for us to reproduce without a save or a level seed.
> 
> Thanks and sorry for the inconvenience!

Yes, I am aware that I can break those spikes. Here's a more detailed picture.
![generation](https://user-images.githubusercontent.com/79550132/133087796-71f02892-1ef9-4977-ad72-0b912f30d892.png)

Sorry for the dumb edit. I circled the inaccessible room with red colour and with blue that is the ore I have to mine.

**Comment by JoneKone** [comment]
Can confirm, sometimes this happens.

**Comment by Regalis11** [comment]
Thank you for the extra info! Should be fixed in https://github.com/Regalis11/Barotrauma-development/commit/00c41ad47ca2bd93c739f362f66a8890a3c51d44: the problem was that the mineral missions spawned the minerals on a random edge of a wall chunk that's next to a cave, without taking into account whether that edge is actually facing towards the cave, which sometimes led to this problem if there happened to be empty space behind the wall.
